File: c++/TripService.cpp
#include <list>
#include "packages/TripServiceSupport.h"

UserSession *UserSession::oneUserSession=0;

std::list<Trip> TripService::GetTripsByUser( User *user )
{
    std::list<Trip> triplist ;
    User* loggedUser = UserSession::GetInstance()->GetLoggedUser();
    bool isFriend = false;
    if ( loggedUser )
    {
        std::list<User>::iterator i;
        for( i = user->GetFriends().begin(); i != user->GetFriends().end(); ++i )
        {
            if ( *i == *loggedUser )
            {
                isFriend = true;
                break;
            }
        }
        if (isFriend)
        {
            triplist = TripDAO::FindTripsByUser(user);
        }
        return triplist;
    }
    else
    {
        throw "UserNotLoggedInException";
    }
}

File: dart/lib/trip_service.dart
import 'package:trip_service/trip_service_error_type.dart';
import 'package:trip_service/trip.dart';
import 'package:trip_service/user.dart';
import 'package:trip_service/trip_dao.dart';
import 'package:trip_service/user_session.dart';

class TripService {
  Future<List<Trip>?> getTripsByUser(User user) async {
    List<Trip>? tripList;
    User? loggedUser = await UserSession.instance.getLoggedUser();

    bool isFriend = false;

    if (loggedUser != null) {
      for (User friend in user.getFriends()) {
        if (friend == loggedUser) {
          isFriend = true;
          break;
        }
      }
      if (isFriend) {
        tripList = await TripDAO.findTripsByUser(user);
      }
      return tripList;
    } else {
      throw TripServiceErrorType.userNotLoggedIn;
    }
  }
}
